compiler/qsc_qasm3/src/compile.rs

```diff
@@ -0,0 +1,4363 @@
+        let loop_var_sem_ty =
+            self.get_semantic_type_from_scalar_type(&loop_var_scalar_ty, false)?;
+        let qsharp_ty =
+            self.convert_semantic_type_to_qsharp_type(&loop_var_sem_ty, loop_var.syntax())?;
+
+        let loop_var_symbol = Symbol {
+            name: loop_var.to_string(),
+            span: loop_var_span,
+            ty: loop_var_sem_ty.clone(),
+            qsharp_ty: qsharp_ty.clone(),
+            io_kind: IOKind::Default,
+        };
+
+        self.symbols.push_scope(crate::symbols::ScopeKind::Block);
+        if self.symbols.insert_symbol(loop_var_symbol.clone()).is_err() {
+            self.push_redefined_symbol_error(loop_var.to_string(), loop_var_span);
+            return None;
+        }
+
+        let body = if let Some(stmt) = for_stmt.stmt() {
+            let stmt = self.compile_stmt(&stmt);
+            self.symbols.pop_scope();
+            build_stmt_wrapped_block_expr(stmt?)
+        } else if let Some(block) = for_stmt.body() {
+            let block = self.compile_block_expr(&block);
+            self.symbols.pop_scope();
+            block
+        } else {
+            let span = span_for_syntax_node(for_stmt.syntax());
+            let kind = SemanticErrorKind::ForStatementsMustHaveABodyOrStatement(span);
+            self.push_semantic_error(kind);
+            None?
+        };
+
+        Some(ast_builder::build_for_stmt(
+            &loop_var_symbol,
+            iterable,
+            body,
+            stmt_span,
+        ))
+    }
+
+    fn compile_for_iterable(
+        &mut self,
+        for_iterable: &oq3_syntax::ast::ForIterable,
+    ) -> Option<QasmTypedExpr> {
+        if let Some(expr) = for_iterable.set_expression() {
+            let expr_list = expr.expression_list()?;
+
+            let expression_list = self
+                .compile_expression_list(&expr_list)?
+                .into_iter()
+                .map(|e| e.expr)
+                .collect();
+
+            let expr = build_expr_array_expr(expression_list, span_for_syntax_node(expr.syntax()));
+            Some(QasmTypedExpr {
+                ty: Type::Set,
+                expr,
+            })
+        } else if let Some(expr) = for_iterable.range_expr() {
+            self.compile_range_expr(&expr, for_iterable.syntax())
+        } else if let Some(expr) = for_iterable.for_iterable_expr() {
+            // For iterating over something like bit[n]
+            self.compile_expr(&expr)
+        } else {
+            let span = span_for_syntax_node(for_iterable.syntax());
+            let kind = SemanticErrorKind::ForIterableInvalidExpression(span);
+            self.push_semantic_error(kind);
+            None
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
+    fn compile_gate_decl(&mut self, gate: &oq3_syntax::ast::Gate) -> Option<ast::Stmt> {
+        let name = gate.name()?;
+
+        //let angle_ty = Type::Angle(None, IsConst::True);
```

Comment:
Old Code?

```suggestion
```

Reply:
It isn't old code, it is a note to future self that the type really should be the commented out line, but `Angle` isn't supported yet. I'll update the comment.